src/js/components/Preview.react.js

```diff
@@ -47,92 +57,205 @@ class Preview extends React.Component<Props, State> {
     this.setState({ previewLoading: false });
   };
 
-  sourceLoading = () => {
-    this.setState({ sourceLoading: true });
+  getWSURL = (pathname: string) => {
+    const url = new URL(webserver.baseUrl);
+    url.pathname = pathname;
+    url.search = `url=${encodeURIComponent(this.props.editor.url)}`;
+    return url;
   };
-  sourceFinishedLoading = () => {
-    this.setState({ sourceLoading: false });
+
+  getWSURLParams = () => {
+    return (
+      'rules=' +
+      encodeURIComponent(
+        JSON.stringify(
+          RuleExporter.export(this.props.rules, this.props.settings)
+        )
+      )
+    );
   };
 
   reloadTabBrowserWindow = (
     tabIndex: number,
     browserWindow: BrowserWindow,
-    contentPathname: string,
+    contentPathname: ?string,
     viewSource: boolean,
-    validateRequiredFields: boolean
+    contentHtml: ?string
   ): void => {
-    const url = new URL(webserver.baseUrl);
-    url.pathname = contentPathname;
-
     if (this.state.activeTab == tabIndex && browserWindow != null) {
-      if (validateRequiredFields) {
-        const generatedRules = RuleExporter.export(
-          this.props.rules,
-          this.props.settings
-        );
-        const hasRequiredField = !!(
-          generatedRules &&
-          generatedRules.rules.find(rule => rule.class === 'GlobalRule')
-        );
-
-        if (!hasRequiredField) {
-          browserWindow.loadURL(url.href);
-          return;
+      if (contentPathname && viewSource) {
+        const url = this.getWSURL(contentPathname);
+        const href = (viewSource ? 'view-source:' : '') + url.href;
+        browserWindow.loadURL(href, {
+          postData: [
+            {
+              type: 'rawData',
+              bytes: Buffer.from(this.getWSURLParams()),
+            },
+          ],
+          extraHeaders: 'Content-Type: application/x-www-form-urlencoded',
+        });
+      } else {
+        if (contentHtml) {
+          browserWindow.loadURL(
+            `data:text/html;charset=utf-8,${encodeURIComponent(contentHtml)}`
```

Comment:
This is great! I never managed to get it to work in the past. Perhaps we can (in a future PR) return from the webserver some HTML content that is already formatted to display HTML code that would enable us to do something similar for the source view.